civis/io/_files.py

```diff
@@ -49,6 +60,135 @@ def _get_aws_error_message(response):
     return msg
 
 
+def _buf_len(buf):
+    if hasattr(buf, '__len__'):
+        return len(buf)
+
+    if hasattr(buf, 'len'):
+        return buf.len
+
+    if hasattr(buf, 'fileno'):
+        try:
+            fileno = buf.fileno()
+        except io.UnsupportedOperation:
+            pass
+        else:
+            return os.fstat(fileno).st_size
+
+    if hasattr(buf, 'getvalue'):
+        # e.g. BytesIO, cStringIO.StringIO
+        return len(buf.getvalue())
+
+    log.warning('Could not determine length of file. Defaulting to single put '
+                'instead of multipart upload. If file is >5GB put will fail.')
```

Comment:
very minor, but putting this log message here is coupling this function to where it is being used.  I think it would be cleaner to call this warning from `file_to_civis`.  i.e.

```python
file_size = _buf_len(...)
if not file_size:
    log.warning(...)
```

Reply:
+1